tiledb/common/thread_pool/thread_pool.h

```diff
@@ -162,7 +404,7 @@ class ThreadPool {
    * @return Status::Ok if the task returned Status::Ok, otherwise the error
    * status is returned
    */
-  Status wait(Task& task);
+  Status wait(ThreadPoolTask* task);
```

Comment:
```suggestion
  Status wait(ThreadPoolTask& task);
```

I think a reference would work as well.